### PR TITLE
{chem}[intel/2017b] GPAW v1.3.0

### PR DESCRIPTION
**Pull request by Micket**
Patches add MKL support and fix dynamic python linking by removing a redundant -L flag.
libxc required only a toolchain version bump.

### CODE REVIEW

File: easybuild/easyconfigs/g/GPAW/GPAW-1.3.0-intel-2017b-Python-2.7.14.eb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+homepage = 'https://wiki.fysik.dtu.dk/gpaw/'
+description = """GPAW is a density-functional theory (DFT) Python code based on the projector-augmented wave (PAW)
+ method and the atomic simulation environment (ASE). It uses real-space uniform grids and multigrid methods or
+ atom-centered basis-functions."""
+
+
+toolchain = {'name': 'intel', 'version': '2017b'}
+
+source_urls = ['https://pypi.python.org/packages/11/59/bfb6102911f9c0c35bfd0c1f3f51e0f4c31523b9f700674d18e4f60d7081/']
+sources = [SOURCELOWER_TAR_GZ]
```
Comment (verdurin) [inline easybuild/easyconfigs/g/GPAW/GPAW-1.3.0-intel-2017b-Python-2.7.14.eb]: Hi @Micket - welcome to `EasyBuild`, and thanks for your contribution.

I'll just add a couple of comments about this PR, to meet the guidelines.

For example, we now ask people to include a checksum for the sources. You can add this automatically with the `--inject-checksums` command-line option. Note that we use `sha256`.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 3 out of 3 (2 easyconfigs in this PR)
rescomp2 - Linux centos 6.8, Intel(R) Xeon(R) CPU E5-2650 v2 @ 2.60GHz, Python 2.7.14
See https://gist.github.com/2d9942f574258d087fb43fab500e874b for a full test report.

File: easybuild/easyconfigs/g/GPAW/GPAW-1.3.0-intel-2017b-Python-2.7.14.eb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+patches = [
+    'GPAW-1.3.0-dynamic_linking.patch',  # removes LIBPL from -L as linker only finds static library in this directory
+    'GPAW-1.3.0-mkl.patch',
+]
+
+python = 'Python'
+pythonver = '2.7.14'
+pythonshortver = '.'.join(pythonver.split('.')[0:2])
+versionsuffix = '-%s-%s' % (python, pythonver)
```
Comment (verdurin) [inline easybuild/easyconfigs/g/GPAW/GPAW-1.3.0-intel-2017b-Python-2.7.14.eb]: You don't actually need to handle the Python dependency this way. There is built-in template support for Python and other languages. See http://easybuild.readthedocs.io/en/latest/version-specific/easyconfig_templates.html

Here's an example showing how it's done:

https://github.com/easybuilders/easybuild-easyconfigs/blob/master/easybuild/easyconfigs/p/pandas/pandas-0.20.1-intel-2017a-Python-3.6.1.eb

File: easybuild/easyconfigs/g/GPAW/libxc-3.0.0-intel-2017b.eb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+homepage = 'http://www.tddft.org/programs/octopus/wiki/index.php/Libxc'
+description = """Libxc is a library of exchange-correlation functionals for density-functional theory.
+ The aim is to provide a portable, well tested and reliable set of exchange and correlation functionals."""
+
+toolchain = {'name': 'intel', 'version': '2017b'}
+# Results for some functionals (e.g. mgga_c_tpss) deviate with too aggressive optimization settings.
+toolchainopts = {'lowopt': True}
+
+sources = [SOURCE_TAR_GZ]
+source_urls = ['http://www.tddft.org/programs/octopus/down.php?file=libxc/']
```
Comment (verdurin) [inline easybuild/easyconfigs/g/GPAW/libxc-3.0.0-intel-2017b.eb]: Again, please add an `sha256` checksum here.

**Comment by Micket** [comment]
Hi verdurin,

A small clarification; does %(pyver)s only work for the version suffix? Because in the example you link, the python verison '3.6.1' is still typed out all the same, just that it's directly specified in the dependencies.

**Comment by Micket** [comment]
I realize that i put the libxc in the wrong directory, and that there was already a libxc 3.0.0 version in the development branch. I removed my version.

**Comment by verdurin** [comment]
@Micket - yes, you still need to spell out the `Python` version. I found that jarring at first, but it does make sense.

**Comment by akesandgren** [comment]
Compare with #5281

**Comment by migueldiascosta** [comment]
can this be closed now that #6984 is merged?

**Pull request closed without merging**